com.unity.netcode.gameobjects/Runtime/Messaging/Messages/ChangeOwnershipMessage.cs

```diff
@@ -0,0 +1,53 @@
+namespace Unity.Netcode.Messages
+{
+    internal struct ChangeOwnershipMessage : INetworkMessage
+    {
+        public ulong NetworkObjectId;
+        public ulong OwnerClientId;
+
+        public void Serialize(ref FastBufferWriter writer)
+        {
+            writer.WriteValueSafe(this);
```

Comment:
Just gonna comment this once here but it applies to most of the IMessages here. By just doing a block copy here we are wasting a lot of bandwidth which in my opinion is not worth the performance gain. The client and NetworkObjectIds are usually very small numbers and profit heavily from packed writes. So I'd really suggest to manually serialize the properties using packed writes for messages like this.

Reply:
I didn't avoid packing for performance reasons, I avoided it because this is coming in really hot and including packing would be one more piece that could break right before release. I thought it would be better to reduce risk for this release by leaving it out, and go back and add packing back in when things calm down again.

Reply:
Though when we do add packing in, I encourage a thought process on the performance vs. bandwidth topic. I don't think everything should be packed. I think we should focus on packing the things that get us the greatest bandwidth savings. If we're saving one byte 50% of the time by packing something, I would favor performance in that case.

Reply:
Ugh fine 😄. But for the record I do strongly disagree. CPU should not be a concern here. Packing is always worth it, we are very much bandwidth constrained at the moment.